StripeFinancialConnections/StripeFinancialConnections/Source/Native/AccountPicker/AccountPickerViewController.swift

```diff
@@ -387,11 +387,52 @@ final class AccountPickerViewController: UIViewController {
                         selectedAccounts = self.dataSource.selectedAccounts
                     }
 
-                    self.delegate?.accountPickerViewController(
-                        self,
-                        didSelectAccounts: selectedAccounts,
-                        nextPane: linkedAccounts.nextPane
-                    )
+                    // for data flows (external_api), where the user is already
+                    // determined to be a Link consumer, we need to take an
+                    // additional step of saving the accounts to Link
+                    if
+                        self.dataSource.manifest.isNetworkingUserFlow == true,
+                        // make sure the current user is a Link consumer
+                        self.dataSource.manifest.accountholderIsLinkConsumer == true,
+                        // make sure this is not a payment flow; in payment flows,
+                        // the AttachLinkedPaymentAccount handles saving to link
+                        !self.dataSource.manifest.shouldAttachLinkedPaymentMethod,
+                        selectedAccounts.count > 0,
+                        let consumerSessionClientSecret = self.dataSource.consumerSessionClientSecret
+                    {
+                        self.dataSource.saveAccountsToLink(
+                            consumerSessionClientSecret: consumerSessionClientSecret
+                        )
```

Comment:
[equivalent web code](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/ba55e4ac/src/linkedAccounts/inner/hooks/v3/useSaveToNetworkAndLink.ts#L101-L122)

(note that this is missing `pollAccountNumbersForSelectedAccounts` which will be tackled in another PR)